cpp/src/arrow/dataset/dataset_writer_test.cc

```diff
@@ -198,9 +198,7 @@ class DatasetWriterTestFixture : public testing::Test {
       int num_batches = 0;
       AssertBatchesEqual(*MakeBatch(expected_file.start, expected_file.num_rows),
                          *ReadAsBatch(written_file->data, &num_batches));
-      if (check_num_record_batches) {
```

Comment:
Can we remove `bool check_num_record_batches = true` argument?

Reply:
Sure, this is added in my previous patch https://github.com/apache/arrow/pull/38885 . Here if not passing `check_num_record_batches = false`, the check would failed because zero-sized batch.

In this patch, it would not be produced, so I can remove it

Reply:
OK. Please remove it before we merge this.